### PR TITLE
Ajaxing

### PR DESCRIPTION
**Pull request by Elffers**

### CODE REVIEW

**Comment by Elffers** [comment]
Still working on specs for resque and mailer, not sure if they're working properly? (in posts_controller_spec.rb, post_mailer_spec.rb
@drbrain @bookis

**Comment by lizrush** [comment]
I'm going to merge this but we still would like code review on the specs!

File: spec/controllers/events_controller_spec.rb [hunk 1]
```diff
@@ -355,6 +368,24 @@
           get :rsvp, id: event.id
           expect(event.users).to include participant
         end
+
+        it 'sends email to user' do
+          without_resque_spec do
+            get :rsvp, id: event.id
+            RsvpMailer.confirmation(event.id, participant.id).deliver
+            expect(ActionMailer::Base.deliveries).to_not be_empty
```
Comment (bookis) [inline spec/controllers/events_controller_spec.rb:376]: You may want to consider testing the exact number of elements within the `deliveries` array. Just to ensure that multiple emails aren't being created and/or the array doesn't just have residual element from some other process.

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -107,18 +111,20 @@
...
-        xit 'sends an email to users who want email update' do
-          post :create, post: valid_attributes
-          # p ActionMailer::Base.deliveries
-          # p "MAIL", PostMailer.new_post(assigns(:post).id, assigns(:user).id).deliver
-          expect(ActionMailer::Base.deliveries).to_not be_empty
+        it 'assigns correct email recipients' do
+          without_resque_spec do
+            post :create, post: valid_attributes
+            expect(assigns(:recipients)).to_not include unsubscribed_user
+            expect(assigns(:recipients)).to include user
```
Comment (bookis) [inline spec/controllers/posts_controller_spec.rb:126]: It's a general best practice to only put one expectation per test.  Line 125 & 126 could be separated in to two separate specs one for "includes subscribed users", and one for "does not include unsubscribed user".

File: spec/controllers/posts_controller_spec.rb [hunk 3]
```diff
@@ -128,7 +134,13 @@
 
         it "adds Postmailer.new_post to the Email queue" do
           post :create, post: valid_attributes
-          EmailJob.should have_queued(assigns(:post).id, assigns(:user).id)
+          # p ResqueSpec.queue_by_name(:email)
+          EmailJob.should have_queued(assigns(:post).id, user.id)
```
Comment (bookis) [inline spec/controllers/posts_controller_spec.rb:138]: I've never used these Resque spec matchers (not sure if it breaks for some reason), but using `expect` is preferred to `should`.

**Comment by bookis** [comment]
This code is exceptional, nice job. Any specific code you'd like more feedback on?

**Comment by Elffers** [comment]
I was mostly concerned that the tests for the action that calls the mailer wasn't necessarily ensuring that emails were being sent to the right users.  It still sometimes feels like I'm designing the test data objects specifically so that they will pass the tests, but that the code itself isn't being tested properly. Specifically, it feels that taking out the resque spec and forcing the rspec to run the mailer action is like testing to see if true equals true.

these tests will change a bit once I finish refactoring the user's preferences to be an hstore hash, but regardless I wanted to make sure I was setting up the tests properly to test that emails were being sent to the correct recipients.

File: spec/controllers/events_controller_spec.rb (hunk 1 above)
Comment (drbrain) [inline spec/controllers/events_controller_spec.rb:376]: I prefer to create a separate RsvpJob test to make sure that the emails get delivered with all the right things inside.  Testing  mail delivery from the controller turns it into more of a functional test.

**Pull request merged**